nixos/modules/virtualisation/containers.nix

```diff
@@ -518,6 +518,7 @@ in
         KillSignal = "WINCH";
 
         DevicePolicy = "closed";
+        DeviceAllow = "/dev/net/tun rw";
```

Comment:
I opened an [issue](https://github.com/NixOS/nixpkgs/issues/18838) regarding this, but I currently lack of a good idea, how this could be solved properly in future.